tests/unit/test_resolver.py

```diff
@@ -93,3 +97,71 @@ def fetch(req: Request) -> RequestOutput:
         assert resolver.resolve_packages(request) == COMBINED_OUTPUT
 
     assert calls_by_pkgtype == ["gomod", "npm", "pip"]
+
+
+@pytest.mark.parametrize(
+    "flags",
+    [
+        pytest.param(["dev-mode"], id="dev-mode-true"),
+        pytest.param([], id="dev-mode-false"),
+    ],
+)
+def test_dev_mode(flags: list[str]) -> None:
+    with mock.patch.dict(
+        resolver._dev_package_managers,
+        values={"shrubbery": lambda _: RequestOutput.empty()},
+        clear=True,
+    ):
+
+        class foo:
+            type: str
+
+        dev_package_input = foo()
+        dev_package_input.type = "shrubbery"
+
+        request = mock.Mock()
+        request.flags = flags
+        request.packages = [dev_package_input]
+
+        if flags:
+            assert resolver.resolve_packages(request) == RequestOutput(
+                components=[], build_config=BuildConfig(environment_variables=[], project_files=[])
+            )
+        else:
+            expected_error = re.escape("Package manager(s) not yet supported: shrubbery")
+            try:
+                resolver.resolve_packages(request)
+            except UnsupportedFeature as exc:
+                assert re.match(expected_error, str(exc))
+
+
+def test_dev_mode_with_released() -> None:
+    with (
+        mock.patch.dict(
+            resolver._package_managers,
+            values={"gomod": lambda _: RequestOutput.empty()},
+            clear=True,
+        ),
+        mock.patch.dict(
+            resolver._dev_package_managers,
+            values={"pip": lambda _: RequestOutput.empty()},
+            clear=True,
+        ),
+    ):
+
+        class foo:
+            type: str
+
+        dev_package_input = foo()
```

Comment:
We could make this a `mock.Mock()` and drop the foo class, and the code would still work as expected.

Reply:
Must've been an in-progress change which cause the `mock.Mock()` version to fail (something about `Mock` not being iterable). Works now.